src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2106,14 +2121,45 @@ int wm_vuldet_report_agent_vulnerabilities(sqlite3 *db, agent_software *agent, w
     return retval;
 }
 
-int wm_vuldet_check_agent_vulnerabilities(agent_software *agents, wm_vuldet_flags *flags, time_t min_full_scan_interval) {
-    agent_software *agents_it;
+
+int wm_vuldet_find_obsolete_vulnerabilities(scan_ctx_t* scan_ctx) {
+    int sock = wm_vuldet_get_wdb_socket();
+    cJSON* j_obsolete_vulns = NULL;
+    cJSON *j_vuln = NULL;
+
+    if (scan_ctx->scan_type == VU_BASELINE_SCAN) {
+        return OS_SUCCESS;
+    }
+
+    j_obsolete_vulns = wdb_agents_vuln_cves_remove_by_status(scan_ctx->agent_id, VULN_CVES_STATUS_PENDING, &sock);
+    if (!j_obsolete_vulns) {
```

Comment:
Is there any possibility of not having vulnerabilities in pending status but yes in obsolete status? I'm wondering whether we should continue the execution of the method without returning here.

Reply:
(!j_obsolete_vulns) represent that the WazuhDB query wasn't executed successfully. Otherwise, an empty JSON array (**[]**) will be returned.
I think we should handle this kind of error by breaking the scan, same as other parts of Vulnerability Detector.

Reply:
Perfect!